### DISCUSSION TITLE
.jp2 files not generating

### DISCUSSION THREAD

**Discussion by Tim-Healey** [NoahGWood/OpenEFT#18] [Q&A] [answered]
On Step 2 nfseg fails with non-zero exit status 254.
![image](https://user-images.githubusercontent.com/1662817/230131327-3941b502-7dab-4b58-91ff-6ed52dace14c.png)
.PNG images of left 4, right 4, and thumbs are found in the directory, but the .jp2 files are not created.
![image](https://user-images.githubusercontent.com/1662817/230131639-131a776b-7c6b-401c-8007-fc01a76e9eef.png)
I have re-installed OpenJPEG.
Has anyone else run into this problem?

**Comment by jstrahl3** [comment]
After getting the NIST stuff compiled/installed, did you add the directory to your PATH?  I forgot that step too and got a similar error.  Sorry, I don't have the link in front of me, but it was easy to google.  The "install_path/bin" is what needs added

**Reply by Tim-Healey** [reply]
I run `export PATH="/home/tim/Documents/OpenEFT/build/bin:$PATH"`(my install directory). I've also added the same thing to ~/.bashrc.  I am able run the command nfseg from a new terminal instance. The program is looking specifically for a .jp2 file which doesn't seem to be created in the tmp directory when openeft.py is run.

**Reply by tjordan170** [reply]
Did you add openjpeg to your path as well?

**Reply by Tim-Healey** [reply]
I believe so, it is installed to `/usr/bin` which is added to my path and I can run commands like `opj_compress` from a new terminal.

**Comment by Tim-Healey** [comment, marked as answer]
Alright, the issue was that OpenJPEG had been installed without png library support.

I reinstalled OpenJPEG to '/usr/local/bin' using `-DBUILD_THIRDPARTY:bool=on` which installs using third party libraries that support png. Then I added it to my path. I think I finally have OpenEFT working correctly! Thanks everyone for their help.

One question, when I've generated my .EFT file, I only have the Right Four fingers, Left four fingers, and 2 thumbs. Individual fingerprints are not shown. Is this the same for everyone else?

**Reply by NoahGWood** [reply]
`One question, when I've generated my .EFT file, I only have the Right Four fingers, Left four fingers, and 2 thumbs. Individual fingerprints are not shown. Is this the same for everyone else?`

Yeah, that's correct. The ATF's eForms portal only supports the slap and two thumbprints. If you wanted to add all of the fingers, you can do that easily by uncommenting them in the fd258_ocr.py file located in the conversion/core directory:
![prints](https://user-images.githubusercontent.com/26890744/230263347-16900177-d1f9-4a38-be4c-2bd075ab5851.png)

Like I said though, for some reason the ATF only supports those three
